src/biogeophys/SoilStateType.F90

```diff
@@ -384,17 +384,20 @@ subroutine Restart(this, bounds, ncid, flag)
     call restartvar(ncid=ncid, flag=flag, varname='SMP', xtype=ncd_double,  &
          dim1name='column', dim2name='levgrnd', switchdim=.true., &
          long_name='soil matric potential', units='mm', &
+         scale_by_thickness=.true., &
```

Comment:
@billsacks in response to your question [here](https://github.com/ESCOMP/CTSM/pull/1301#pullrequestreview-630252567):

There are four variables that get `scale_by_thickness=.true.`.
- SMP
- HK
- H2OSOI_LIQ
- H2OSOI_ICE

Reply:
Are those really the only ones that need it?

- [ ] What about all of the BGC variables?

- [ ] Also, it seems like, in principle, some 2-d variables in accumulMod could need this. For example, what if you are accumulating the 10-day running average of H2OSOI_LIQ: then you would want scale_by_thickness to be true for that accumulated variable, too, right? It seems like handling this correctly would require passing scale_by_thickness into `init_accum_field`: that could be an optional argument that is required if numlev > 1.

Reply:
> Are those really the only ones that need it?
> 
> * [ ]  What about all of the BGC variables?

BGC variables have units of g/m3, so they are fine if I'm thinking about this correctly.

> * [ ]  Also, it seems like, in principle, some 2-d variables in accumulMod could need this. For example, what if you are accumulating the 10-day running average of H2OSOI_LIQ: then you would want scale_by_thickness to be true for that accumulated variable, too, right? It seems like handling this correctly would require passing scale_by_thickness into `init_accum_field`: that could be an optional argument that is required if numlev > 1.

I didn't look through the accumulMod list of variables, so I definitely missed those. Thanks for catching that! I will look there next.

Reply:
> BGC variables have units of g/m3, so they are fine if I'm thinking about this correctly.

@billsacks LMK if you'd like me to revisit this, but I feel I looked at these carefully. I should mention: I did come across BGC vars with units of g/m2, but not among the ones that we send to restart.

> I didn't look through the accumulMod list of variables, so I definitely missed those. Thanks for catching that! I will look there next.

I think I made the right changes here. I didn't find accumulation vars with numlev > 1, but I introduced the necessary code. Unit tests still pass and my sample test
`ERP_D_Ld3_P36x2.f10_f10_mg37.I2000Clm50BgcCru.cheyenne_intel.clm-default`
also still passes. And with slight modifications the latter test still shows bfb against baseline.

Reply:
Regarding BGC vars: good point; thanks.